### ISSUE TITLE
Esbuild ouput target does not match projects tsconfig target

### ISSUE THREAD

**Issue by SyedRFarhan** [serverless-stack/serverless-stack#107]
I've discovered this issue when using optional chaining in my typescript code which is set to transpile down to ES2015 code in my tsconfig. However esbuild does not use this target so the actual output code is ES2020 or whatever default target esbuild has which includes optional chaining in the js bundle. This causes errors when invoke my lambda because Node 12 does not support optional chaining. 

So a potential solution could be to read the projects tsconfig target and pass it into esbuild or allow for a target to be set via some other config.

Here's a repo I created with the issue recreated: https://github.com/SyedRFarhan/sst-esbuild-target-mismatch
Run this with Node 12 and invoke the function to recreate my issue.

**Comment by fwang** [comment]
@SyedRFarhan Thanks for reporting this. Can you share the step by step commands I need to run after git cloning your repo to reproduce the issue?

**Comment by SyedRFarhan** [comment]
- Set your environment to Node 12
- npm run start
- Make a get request to the lambda endpoint
- An error should be printed to the terminal regarding the use of optional chaining syntax

**Comment by fwang** [comment]
Thanks @SyedRFarhan

Can you confirm this is the error you are getting on your side?

```
{ name: 'SyntaxError',
  message: 'Unexpected token .',
  stack:
   '/Users/frank/Sites/fwang/sst-esbuild-target-mismatch/.build/src/lambda.js:16
     return someObject?.anything ?? "does not exist";
                       ^
   
   SyntaxError: Unexpected token .
       at Module._compile (internal/modules/cjs/loader.js:723:23)
       at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
       at Module.load (internal/modules/cjs/loader.js:653:32)
       at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
       at Function.Module._load (internal/modules/cjs/loader.js:585:3)
       at Module.require (internal/modules/cjs/loader.js:692:17)
       at require (internal/modules/cjs/helpers.js:25:18)
       at getHandler (/Users/frank/Sites/fwang/sst-esbuild-target-mismatch/node_modules/@serverless-stack/cli/assets/lambda-invoke/bootstrap.js:87:15)
       at start (/Users/frank/Sites/fwang/sst-esbuild-target-mismatch/node_modules/@serverless-stack/cli/assets/lambda-invoke/bootstrap.js:32:15)
       at Object.<anonymous> (/Users/frank/Sites/fwang/sst-esbuild-target-mismatch/node_modules/@serverless-stack/cli/assets/lambda-invoke/bootstrap.js:26:1)'
}
```

**Comment by SyedRFarhan** [comment]
@fwang

Yes, that's the error.

**Comment by jayair** [comment]
@SyedRFarhan Just had a look at this. This is a bug on our side. We are not explicitly setting the esbuild target right now. That said, I don't think esbuild uses the target set in the `tsconfig.json`. To ensure compatibility, we'll set the target globally to ES2015 for now.

https://github.com/evanw/esbuild/issues/277

**Comment by ryanmalesic** [comment]
Another quick fix is to use NodeJs 14.x instead of 10 or 12 (if possible) in the function runtime and in the dev environment. With this runtime, the new operators work in the handler at runtime

**Comment by jayair** [comment]
@ryanmalesic Yeah it does. Just tested this as well.

**Comment by fwang** [comment]
@SyedRFarhan This is fixed in release v0.9.5

**Comment by SyedRFarhan** [comment]
Thanks for looking into this!
